packages/twenty-front/src/pages/settings/data-model/SettingsObjectDetail.tsx

```diff
@@ -102,8 +102,12 @@ export const SettingsObjectDetail = () => {
                         onEdit={() =>
                           navigate(`./${getFieldSlug(activeMetadataField)}`)
                         }
-                        onDisable={() =>
-                          disableMetadataField(activeMetadataField)
+                        onDisable={
+                          activeMetadataField.id ===
+                            activeObjectMetadataItem.labelIdentifierFieldMetadataId ||
+                          activeMetadataField.name === 'name'
```

Comment:
I think it's dangerous to rely on "name" here, I'm not sure we want to keep that in the future and we probably want to allow users to define their own labelIdentifier whilst keeping name as an independent field?
wdyt @charlesBochet ?